src/components/CalendarPopover/CalendarPopover.tsx

```diff
@@ -64,9 +71,48 @@ export class CalendarPopover extends React.Component<CalendarPopoverProps> {
       [POPOVER_DATA_KEYS.WithShadow]: withShadow,
       [POPOVER_DATA_KEYS.Animated]: animated,
       [POPOVER_DATA_KEYS.Shown]: isShown,
+      [POPOVER_DATA_KEYS.ManualFocus]: manualFocus,
     };
   };
 
+  _focusLastActive = () => {
+    !this.props.manualFocus &&
+      this.lastActiveElement &&
+      this.lastActiveElement.focus();
+  };
+
+  _onEsc = e => {
+    if (e.keyCode === KEY_CODES.Esc) {
+      this.props.onClose();
+    }
+  };
+
+  manageA11y = () => {
+    if (this.props.isShown) {
+      document.addEventListener('keyup', this._onEsc);
+      this.lastActiveElement = document.activeElement;
+      !this.props.manualFocus && this.iconRef.current.focus();
+    }
+  };
+
+  componentDidUpdate(prevProps) {
+    if (!prevProps.isShown) {
+      this.manageA11y();
+    } else if (prevProps.isShown && !this.props.isShown) {
```

Comment:
Actually `prevProps.isShown` isn't necessary here (because of the `if-else`)
```suggestion
    } else if (!this.props.isShown) {
```
though it's really not critical